token-swap/program/src/curve/constant_price.rs

```diff
@@ -346,4 +282,68 @@ mod tests {
         assert_eq!(result.source_amount_swapped, token_b_price);
         assert_eq!(result.destination_amount_swapped, 1u128);
     }
+
+    proptest! {
+        #[test]
+        fn pool_token_conversion_a_to_b(
+            // in the pool token conversion calcs, we simulate trading half of
+            // source_token_amount, so this needs to be at least 2
+            source_token_amount in 2..u64::MAX,
+            swap_source_amount in 1..u64::MAX,
+            swap_destination_amount in 1..u64::MAX,
+            token_b_price in 1..u64::MAX,
+        ) {
+            let traded_source_amount = source_token_amount / 2;
+            // Make sure that the trade yields at least 1 token B
+            prop_assume!(traded_source_amount / token_b_price >= 1);
+            // Make sure there's enough tokens to get back on the other side
+            prop_assume!(traded_source_amount / token_b_price <= swap_destination_amount);
+
+            let curve = ConstantPriceCurve {
+                token_b_price,
+            };
+            check_pool_token_conversion(
+                &curve,
+                source_token_amount as u128,
+                swap_source_amount as u128,
+                swap_destination_amount as u128,
+                TradeDirection::AtoB,
+                CONVERSION_BASIS_POINTS_GUARANTEE,
+            );
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn pool_token_conversion_b_to_a(
+            // in the pool token conversion calcs, we simulate trading half of
+            // source_token_amount, so this needs to be at least 2
+            source_token_amount in 2..u64::MAX,
+            swap_source_amount in 1..u64::MAX,
+            swap_destination_amount in 1..u128::MAX, // NOTE: this is bumped up to avoid too many proptest rejections
+            token_b_price in 1..u64::MAX,
+        ) {
+            let curve = ConstantPriceCurve {
+                token_b_price,
+            };
+            // The constant price curve needs to have enough destination amount
+            // on the other side to complete the swap
+            let token_b_price = token_b_price as u128;
+            let source_token_amount = source_token_amount as u128;
+            let swap_source_amount = swap_source_amount as u128;
+            prop_assume!(token_b_price * source_token_amount <= swap_destination_amount);
+
+            // basis points guarantee, much higher than other calcs due to potential
+            // truncation from huge token b prices
+            let basis_points_guarantee = 2000;
```

Comment:
I think the bounds need to be tightened to bring this down, might play with this more.